plugins/modules/system/parted.py

```diff
@@ -616,6 +625,10 @@ def main():
         if current_device['generic'].get('table', None) != label:
             script += "mklabel %s " % label
 
+        # parted <= 3.2.153 bug: optional filesystem parameter is mandatory for negative part_start
+        if fs_type is None and part_start.startswith('-'):
+            fs_type = 'ext2'
```

Comment:
Should the default fstype be same whether partition is created from beginning of disk or end ? Also, if we have to keep it different (not an expert on `parted`), maybe we should add this info to docs somewhere ?

Reply:
fs_type is converted to "partition type" by parted on some labels (like: msdos, where is translated to 0x83). If there's no partition type specified, parted utility has to assume one (and it is ext2, as I remember by checking in parted's code). So: addidng ext2 or omiting has the same effect: it is only forced for negative beginnings as a workaround for parted's bug.

So specyfying 'ext2' as fs_type is a no-op (at least for current parted version), but for sake for clarity I could also:
* make 'ext2' a default vaule for fs_type
* get rid of this code completely, or warn in docs that fs_type parameter is sometimes necessary for parted.

#221 is merged only recently so having default value for this would be nice, because it would also make this unnecessary.

Reply:
After consideration, I'm in favour of:
* removing this hack (it was written before fs_type parameter was added, so now it has manual workaround: just add `fs_type: ext2` or whatever)
* warning in documentation of fs_type parameter that is necessary sometimes: if not, fist person to use negative boundaries on RHEL6 (or even RHEL7?) would hit this old bug in parted.

What's your opinion?

Reply:
Sounds good to me!